fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -381,18 +381,21 @@ public void createDb(String dbName, Map<String, String> properties) throws DdlEx
             if (fullNameToDb.containsKey(dbName)) {
                 throw new AlreadyExistsException("Database Already Exists");
             } else {
-                id = getNextId();
-                Database db = new Database(id, dbName);
-                unprotectCreateDb(db);
                 String storageVolumeId = "";
-                if (RunMode.allowCreateLakeTable()) {
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
                     String volume = StorageVolumeMgr.DEFAULT;
                     if (properties != null && properties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME)) {
                         volume = properties.remove(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME);
                     }
-                    storageVolumeId = bindStorageVolumeToDb(id, volume);
+                    storageVolumeId = getStorageVolumeId(volume);
                 }
+                id = getNextId();
+                Database db = new Database(id, dbName);
+                unprotectCreateDb(db);
                 GlobalStateMgr.getCurrentState().getEditLog().logCreateDb(db, storageVolumeId);
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                    GlobalStateMgr.getCurrentState().getStorageVolumeMgr().bindDbToStorageVolume(storageVolumeId, id);
```

Comment:
what if the storageVolume exists in line 390, but removed in other threads when running to line 397?